### ISSUE TITLE
Incomplete error message

### ISSUE THREAD

**Issue by eschnett** [openPMD/openPMD-api#1028]
While debugging the Julia wrapper of openPMD-api, I received the error message below. It is incomplete, and it does not give any information about the second type.
```
LoadError: [ADIOS2] Trying to access a dataset with wrong type (trying to access dataset with type int64_t, but has type )
```

I would have expected to see a type name there, of if there is somehow no type information, say why it is missing.

**Software Environment**
 - version of openPMD-api: [0.14.0], [this](https://github.com/openPMD/openPMD-api/pull/1025) branch
 - installed openPMD-api via: [Yggdrasil]
 - operating system: [macOS 11.4]
 - machine: [laptop]
 - name and version of Python implementation: [n/a]
 - version of HDF5: [n/a]
 - version of ADIOS1: [n/a]
 - version of ADIOS2: [2.7.1]
 - name and version of MPI: [MPICH 3.4.2]

**Comment by franzpoeschel** [comment]
ADIOS2 reports datatypes as strings, and when a datatype cannot be found (e.g. when a variable is not found), it returns an empty string, which is what you're seeing here.
The error message seems a bit cryptic as a result, I'll look into making this more understandable.

**Comment by franzpoeschel** [comment]
Fixed with #1036
